src/Plutarch/Extra/Boost.hs

```diff
@@ -0,0 +1,58 @@
+{- |
+ Module: Plutarch.Extra.Boost
+ Copyright: (C) Liqwid Labs 2022
+ License: Apache 2.0
+ Maintainer: Koz Ross <koz@mlabs.city>
+ Portability: GHC only
+ Stability: Experimental
+
+ Performance-oriented metaprogramming helpers.
+-}
+module Plutarch.Extra.Boost (
+    -- * Unhoisted list helpers
+    psing,
+    plist,
+
+    -- * CPS-driven matching
+    pmatchList,
+) where
+
+{- | Make a list-like structure from any 'Foldable' of 'Term's. This is
+ unhoisted.
+
+ @since 3.4.0
+-}
+plist ::
```

Comment:
These belong on `Plutarch.Extra.List` I think. Or, if it's only used by the sort, they should go into `Ord` I think.